kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,249 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
     labeled_dataset = torch.utils.data.Subset(dataset, tindices)
     unlabeled_dataset = torch.utils.data.Subset(dataset, uindices)
     return labeled_dataset, unlabeled_dataset
+
+
+def split_by_domains(domain_labels, n_partitions, split_ratios):
```

Comment:
+ some docstrings

Reply:
number of returned subsets = #domains x # splits?

Reply:
Sorry for the misleading name. It is actually split different domains with the same ratios. `balanced_domain_split()` may be better. I will rename the function in the next update.

Reply:
Explanation helps to clarify and identify such gaps so try to do so in the next update.